packages/transformer/src/IModelTransformer.ts

```diff
@@ -411,18 +414,144 @@ export class IModelTransformer extends IModelExportHandler {
 
   private _isSynchronization = false;
 
-  private get _isReverseSynchronization() {
-    return this._isSynchronization && this._options.isReverseSynchronization;
+  /**
+   * A private variable meant to be set by tests which have an outdated way of setting up transforms. In all synchronizations today we expect to find an ESA in the branch db which describes the master -> branch relationship.
+   * The exception to this is the first transform aka the provenance initializing transform which requires that the master imodel and the branch imodel are identical at the time of provenance initialization.
+   * A couple ofoutdated tests run their first transform providing a source and targetdb that are slightly different which is no longer supported. In order to not remove these tests which are still providing value
+   * this private property on the IModelTransformer exists.
+   */
+  private _allowNoScopingESA = false;
+
+  public static noEsaSyncDirectionErrorMessage =
+    "Couldn't find an external source aspect to determine sync direction. This often means that the master->branch relationship has not been established. Consider running the transformer with wasSourceIModelCopiedToTarget set to true.";
+
+  /**
+   * querys the scopes external source and sets aspectId, version and jsonProperties on the provided aspectProps if available.
+   * @param dbToQuery db to run the query on for scope external source
+   * @param aspectProps aspectProps to search for
+   * @param opts opts containing preserveVersionReceived and getJsonProperties. if preserveVersionReceived is true and an aspect is found with the provided aspectProps,
+   * then the version passed as part of the aspectProps will be kept on the aspectProps.
+   * If false or not provided, the version on the aspectProps will be overwritten with the version from the queried aspect.
+   */
+  public static queryScopeExternalSource(
+    dbToQuery: IModelDb,
+    aspectProps: ExternalSourceAspectProps,
+    opts?: { getJsonProperties?: boolean; preserveVersionReceived?: boolean }
+  ): void {
+    const sql = `
+      SELECT ECInstanceId, Version
+      ${opts?.getJsonProperties ? ", JsonProperties" : ""}
+      FROM ${ExternalSourceAspect.classFullName}
+      WHERE Element.Id=:elementId
+        AND Scope.Id=:scopeId
+        AND Kind=:kind
+        AND Identifier=:identifier
+      LIMIT 1
+    `;
+    dbToQuery.withPreparedStatement(sql, (statement: ECSqlStatement) => {
+      statement.bindId("elementId", aspectProps.element.id);
+      if (aspectProps.scope === undefined) return; // return instead of binding an invalid id
+      statement.bindId("scopeId", aspectProps.scope.id);
+      statement.bindString("kind", aspectProps.kind);
+      statement.bindString("identifier", aspectProps.identifier);
+      if (DbResult.BE_SQLITE_ROW !== statement.step()) return;
+      aspectProps.id = statement.getValue(0).getId();
```

Comment:
iirc the original code did this, but let's take this chance to remove mutation of input variables and return a new copy instead

Reply:
I changed it to mutate the input variable intentionally, because you had commented that it was strange we were returning aspectProps. Was the strange part that we used to be returning the below type as opposed to a type of ExternalSourceAspectProps ? 

` private queryScopeExternalSource(
    aspectProps: ExternalSourceAspectProps,
    { getJsonProperties = false } = {}
  ): {
    aspectId?: Id64String;
    version?: string;
    /** stringified json */
    jsonProperties?: string;
  } {`

Reply:
maybe?
After looking at it again, I'm pretty sure it _used_ to mutate, I changed it to return which i think is better. Idr what I considered strange exactly, but showing me the original comment would help me...
I do think returning a subset of the aspect props is ok in general, makes it more widely reusable

Reply:
Okay I'll go back to not mutating then, the original comment was over a call so no records of it.

Reply:
Changed